### PR TITLE
fix(EMS-3439): no PDF - Eligibility - Check your answers - Special characters

### PR DESCRIPTION
**Pull request by Zainzzkk**
## Introduction :pencil2:

This PR fixes an issue where the company name with special characters was not displayed correctly

## Resolution :heavy_check_mark:

* Only sanitise the incoming formBody for eligibility in `updateSubmittedData `
* Updated tests
* Added e2e test

### CODE REVIEW

File: e2e-tests/commands/insurance/check-your-answers-eligibility-summary-list.js [hunk 1]
```diff
@@ -68,19 +69,28 @@ const checkYourAnswersEligibilitySummaryList = ({
 
     cy.assertSummaryListRow(summaryList, fieldId, expectedKey, expectedValue, expectedChangeLinkText);
   },
-  [COMPANY_NAME]: () => {
+  [COMPANY_NAME]: ({ differentCompany = false }) => {
```
Comment (ttbarnes) [inline e2e-tests/commands/insurance/check-your-answers-eligibility-summary-list.js:72]: `differentCompany` is very vague, how about `withSpecialCharacters`?

File: e2e-tests/commands/insurance/complete-and-submit-all-insurance-eligibility-answers.js [hunk 2]
```diff
@@ -1,15 +1,16 @@
 import completeAndSubmitBuyerCountryForm from '../forms';
+import { COMPANIES_HOUSE_NUMBER } from '../../constants/examples';
 
 /**
  * completeAndSubmitAllInsuranceEligibilityAnswers
  * completes and submits insurance eligibility answers until the check your answers page
  */
-const completeAndSubmitAllInsuranceEligibilityAnswers = () => {
+const completeAndSubmitAllInsuranceEligibilityAnswers = ({ companyNumber = COMPANIES_HOUSE_NUMBER }) => {
```
Comment (ttbarnes) [inline e2e-tests/commands/insurance/complete-and-submit-all-insurance-eligibility-answers.js:8]: Missing documentation

File: e2e-tests/constants/examples/companies-house-number-examples.js [hunk 3]
```diff
@@ -1,4 +1,5 @@
 export const COMPANIES_HOUSE_NUMBER = '08989898';
+export const COMPANIES_HOUSE_NUMBER_COMPANY_WITH_SPECIAL_CHARACTER_NAME = '00947741';
```
Comment (ttbarnes) [inline e2e-tests/constants/examples/companies-house-number-examples.js:2]: `COMPANIES_HOUSE_NUMBER_SPECIAL_CHARACTER/S_NAME`? More aligned with e.g `COMPANIES_HOUSE_NUMBER_MULTIPLE_SIC_CODES`


Does the company name have multiple special characters?

File: e2e-tests/insurance/cypress/e2e/journeys/eligibility/check-your-answers/eligibility-check-your-answers-company-name-special-characters.spec.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  });
+
+  it(`should render a ${COMPANIES_HOUSE_NUMBER} summary list row`, () => {
+    checkSummaryList[COMPANIES_HOUSE_NUMBER]({ differentCompany: true });
+  });
+
+  it(`should render a ${COMPANY_NAME} summary list row with special characters`, () => {
+    checkSummaryList[COMPANY_NAME]({ differentCompany: true });
+  });
+});
```
Comment (ttbarnes) [inline e2e-tests/insurance/cypress/e2e/journeys/eligibility/check-your-answers/eligibility-check-your-answers-company-name-special-characters.spec.js:35]: Do we have an E2E test for the company details summary list in eligibility rendering special characters as expected?

File: e2e-tests/constants/examples/companies-house-number-examples.js (hunk 3 above)
Comment (Zainzzkk) [reply e2e-tests/constants/examples/companies-house-number-examples.js:2]: Just one

File: e2e-tests/insurance/cypress/e2e/journeys/eligibility/check-your-answers/eligibility-check-your-answers-company-name-special-characters.spec.js (hunk 4 above)
Comment (Zainzzkk) [reply e2e-tests/insurance/cypress/e2e/journeys/eligibility/check-your-answers/eligibility-check-your-answers-company-name-special-characters.spec.js:35]: Can add one

**Review by ttbarnes — APPROVED** [review]

**Pull request merged**